File: test_youtubei.js
const https = require('https');
const videoId = "inEu2qQuGZ8";
const postData = JSON.stringify({
    context: {
        client: {
            clientName: "TVHTML5_SIMPLY_EMBEDDED_PLAYER",
            clientVersion: "2.0"
        },
        thirdParty: {
            embedUrl: "https://www.youtube.com/watch?v=" + videoId
        }
    },
    videoId: videoId
});
const options = {
    hostname: 'youtubei.googleapis.com',
    path: '/youtubei/v1/player',
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        'Content-Length': postData.length
    }
};
const req = https.request(options, (res) => {
    let data = '';
    res.on('data', (chunk) => { data += chunk; });
    res.on('end', () => {
        if (data.includes('hlsManifestUrl')) {
            console.log("Has hlsManifestUrl:", data.split('hlsManifestUrl":"')[1].split('"')[0]);
        }
        if (data.includes('dashManifestUrl')) {
            console.log("Has dashManifestUrl:", data.split('dashManifestUrl":"')[1].split('"')[0]);
        }
        console.log("Found formats count:", (data.match(/"itag":/g) || []).length);
    });
});
req.write(postData);
req.end();

File: test_youtubei.kt
import java.net.URL
import java.net.HttpURLConnection

fun main() {
    val videoId = "inEu2qQuGZ8"
    val url = URL("https://youtubei.googleapis.com/youtubei/v1/player")
    val connection = url.openConnection() as HttpURLConnection
    connection.requestMethod = "POST"
    connection.setRequestProperty("Content-Type", "application/json")
    connection.doOutput = true
    
    val body = """{
        "context": {
            "client": {
                "clientName": "ANDROID",
                "clientVersion": "17.36.4"
            }
        },
        "videoId": "$videoId"
    }"""
    
    connection.outputStream.write(body.toByteArray())
    connection.connect()

    val response = connection.inputStream.bufferedReader().readText()
    if (response.contains("hlsManifestUrl")) {
        println("Has hlsManifestUrl: " + response.substringAfter("hlsManifestUrl\":\"").substringBefore("\""))
    }
    if (response.contains("dashManifestUrl")) {
        println("Has dashManifestUrl: " + response.substringAfter("dashManifestUrl\":\"").substringBefore("\""))
    }
}
